src/java_bytecode/ci_lazy_methods.cpp

```diff
@@ -196,7 +198,8 @@ bool ci_lazy_methodst::operator()(
       // Don't keep functions that belong to this language that we haven't
       // converted above
       if(
-        method_bytecode.contains_method(sym.first) &&
+        (method_bytecode.contains_method(sym.first) ||
+         synthetic_methods.count(sym.first)) &&
```

Comment:
When is this first condition false for symbol entries that are code (surely it is either a user function or a synthetic function?)

Reply:
This could be read as `is-normal-method || is-synthetic-method`, which I think matches your expectation.

Reply:
Yes - but when is the above condition false?

Reply:
Ah, when the method belongs to a different frontend. For Java that case is currently unsupported, but cross-language or GOTO-binary + Java or similar possibilities are supported by our current architecture, so it would be best not to rule that case out when it's easy to support.

Better summary of this line: "if this method relates to this languaget" (otherwise leave it alone)

Reply:
Ah yes - which matches with the comment :+1: